### ISSUE TITLE
Support Client HTTPS certifcates for private git repositories

### ISSUE THREAD

**Issue by ldue** [argoproj/argo-cd#1945]
In my company our git repositories are only available with tls client authentication. Sadly argocd does not support passing client tls configurations as repository credentials right now.
Workarounds (e.g. a proxy server injecting the certificates) are not so feasible.

I would think providing functionality similar to the handling of SSH keys for repositories like:
````
argocd repo add git@github.com:argoproj/argocd-example-apps.git --tls-client-cert-path ~/.certs/cert.pem --tls-client-cert-key-path ~/.certs/user.key
````

Implementation seems from my superficial judging not to complicated.

**Comment by jannfis** [comment]
I think is a great feature to have. I had a quick glance, the foundation to implement it is there so the change should be marginal (famous last words of every developer).
However, waiting for PR #1941 to be merged or rejected first, since there is some interdependency in the code base that this change will touch.

**Comment by jannfis** [comment]
As #1941 has been merged, now starting work on this issue.

**Comment by jannfis** [comment]
@ldue Feature is ready for testing in `argocd:latest` - it'd be awesome if you could test whether it works for you or not. Thanks!

**Comment by ldue** [comment]
@jannfis thank you very much for your effort you invested! Such a quick implementation! 

i tried to install the version from the master using the install.yaml and got the following issue with the repo-server and the dex-server:
```
goroutine 1 [running]:
github.com/argoproj/argo-cd/util/assets.init.0()
        /go/src/github.com/argoproj/argo-cd/util/assets/assets.go:19 +0x441
```

Is there something special I need to do to use the dev version?

**Comment by jannfis** [comment]
@ldue `latest` builds are unfortunately broken at the moment, please refer to issue #2045 and PR #2041 , which is supposed to fix it. Once PR #2041 (or a different fix) has been merged, the next build iteration of `argocd:latest` should be usable again. Sorry for the inconvinience.

**Comment by ldue** [comment]
Ah okay thank you, nothing to be sorry for.

**Comment by jannfis** [comment]
Just a little heads up, PR has been merged and fixed image was built and pushed. :)

**Comment by ldue** [comment]
Finally got around to test it, works like a charm! 
Big Kudos to you argo guys!

So this will be included in the next release in end of August?

**Comment by jannfis** [comment]
Thanks a lot for testing & feedback! According to the roadmap, this feature will be included in the upcoming v1.2 end of August, but I can't tell for sure.
